api/lib/infrastructure/repositories/knowledge-element-repository.js

```diff
@@ -92,12 +77,49 @@ module.exports = {
     return _.groupBy(knowledgeElements, 'competenceId');
   },
 
-  findByCampaignIdAndUserIdForSharedCampaignParticipation({ campaignId, userId }) {
-    return _findByCampaignIdForSharedCampaignParticipation({ campaignId, userId });
+  async findValidatedByCampaignIdAndUserIdForSharedCampaignParticipationInTargetProfile({ campaignId, userId }) {
+    const sharedCampaignParticipations = await knex('campaign-participations')
```

Comment:
On pourrait déstructurer le tableau ici pour récupérer que la campagne qui nous intéresse. Qu'en pensez-vous ?
```suggestion
    const [sharedCampaignParticipation] = await knex('campaign-participations')
```

Reply:
done